frontend/src/widgets/FrequencyGraph.js

```diff
@@ -0,0 +1,131 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { Grid } from '@trussworks/react-uswds';
+import { capitalize } from 'lodash';
+
+import withWidgetData from './withWidgetData';
+import Container from '../components/Container';
+import AccessibleWidgetData from './AccessibleWidgetData';
+import BarGraph from './BarGraph';
+import ButtonSelect from '../components/ButtonSelect';
+import './FrequencyGraph.css';
+
+const SORT_ORDER = {
+  DESC: 1,
+  ALPHA: 2,
+};
+
+function sortData(data, order) {
+  const sortedData = [...data];
+  if (order === SORT_ORDER.ALPHA) {
+    sortedData.sort((a, b) => a.topic.localeCompare(b.topic));
+  } else {
+    sortedData.sort((a, b) => b.count - a.count);
+  }
+  return sortedData;
+}
+
+const TOPIC = 0;
+const REASON = 1;
+
+const HEADINGS = {
+  [TOPIC]: ['Topic', 'Count'],
+  [REASON]: ['Reason', 'Count'],
+};
+
+export function FreqGraph({ data, loading }) {
+  // whether to show the data as accessible widget data or not
+  const [showAccessibleData, updateShowAccessibleData] = useState(false);
+  const [selectedGraph, updateSelectedGraph] = useState(TOPIC);
+  const selectedGraphString = selectedGraph === TOPIC ? 'topic' : 'reason';
+
+  const selectedData = data[selectedGraphString];
+  const sortedData = sortData(selectedData, SORT_ORDER.DESC);
+  const accessibleRows = sortedData.map((row) => ({ data: [row.category, row.count] }));
+
+  const columnHeadings = HEADINGS[selectedGraph];
+
+  // toggle the data table
+  function toggleType() {
+    updateShowAccessibleData(!showAccessibleData);
+  }
+
+  const onSelectedGraphChange = (e) => {
+    updateSelectedGraph(e.value);
+  };
+
+  return (
+    <Container className="ttahub--frequency-graph" padding={3} loading={loading} loadingLabel={`${selectedGraph} frequency loading`}>
+      <Grid row className="position-relative margin-bottom-2">
+        <Grid className="flex-align-self-center" desktop={{ col: 'auto' }} mobileLg={{ col: 10 }}>
+          <ButtonSelect
```

Comment:
I can select it with the keyboard! That's great.
The reader interaction is close, but I think just needs a tuneup

Right now if I proceed through the document, when I get to the top of this widget, I am read this:
- Open change graph menu, button
- Heading, level 2 - In activity reports by frequency

I think there are two easy solutions
- Expand the button description to include the "activity reports by frequency" bit and then hide the h2 via aria-hidden=true.
- Add some visually hidden text to the heading to explain the full context

I swear I **can** do a screenreader demo if that would be helpful

Reply:
Yeah we might need to pair for that, the system screen reader I have for linux doesn't seem to function properly (I'm pretty sure it doesn't read out the h2 at all).